File: 1004-max-consecutive-ones-iii/1004-max-consecutive-ones-iii.cpp
class Solution {
public:
    int longestOnes(vector<int>& nums, int k) {
        int i=0 ,j=0 ,n=nums.size(),countZero=0, ans=INT_MIN;
            
            while(j<n)
            {
                if(nums[j]==0)
                   countZero++;
            
                    while(countZero>k)
                    {
                        if(nums[i]==0)
                        countZero--;
                        i++;
                    }
                ans=max(ans,(j-i)+1);
                j++;
            }
            return ans;
    }
};

File: 1004-max-consecutive-ones-iii/1004-max-consecutive-ones-iii.java
class Solution {
    public int longestOnes(int[] nums, int k) {
        int s=0, e=0 ,flippedOnes=0, maxLen=0;
            
            while(e<nums.length)
            {
                    if(nums[e]==1)
                    {
                        e++;                            
                    }else{ 
                        //(nums[e]==0)
                        if(flippedOnes<k)
                        {
                                flippedOnes++;
                                e++;
                        }else{
                                //reduce some of flips
                                while(flippedOnes>=k)
                                {
                                        if(nums[s]==0){
                                           flippedOnes--;
                                        }
                                  s++;
                                }                                
                        }
                    }
                    maxLen=Math.max(maxLen,(e-s));
            }            
            return maxLen;
    }        
}